### PR TITLE
fix: 9414 Firestore where clause not fetching numbers

### LINKED ISSUE #9414: [Bug]: Firestore where clause doesn't seem to work
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When querying Firestore, if I add a where clause, it doesn't seem to work. I get an empty dataset back even though I know that the record exists. This seems to happen for integer values.

### Steps To Reproduce

1. Create an app with Firestore query
2. List the documents in the collection
3. Add a where clause filter to search for a specific document
4. See that the response is an empty array when you know that the relevant document exists.

Screenshots below:
Screenshot 2021-11-27 at 9 59 10 AM
Screenshot 2021-11-27 at 9 58 54 AM

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by leotom2000**
## Description

 Fixed issue with Firestore in fetching data while using numbers in the where condition.
 Other Fixes made in the where condition :
1. When using Boolean
2. When using Date
3. When Using Timestamp
4. When given space in the key
5. When Space given in the Value

Fixes #9414
 
## Type of change

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

## How Has This Been Tested?

> Manually

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/utils/WhereConditionUtils.java [hunk 1]
```diff
@@ -37,6 +43,44 @@ public static Query applyWhereConditional(Query query, String path, String opera
             );
         }
 
+        DataType dataType = DataTypeStringUtils.stringToKnownDataTypeConverter(strValue);
+        Object value = strValue.trim();
+
+        switch (dataType) {
```
Comment (sumitsum) [inline app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/utils/WhereConditionUtils.java:49]: Can you please add a test case for integer, boolean, date and timestamp types ?

Comment (leotom2000) [reply app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/utils/WhereConditionUtils.java:49]: Added test Cases

**Comment by sumitsum** [comment]
LGTM. Requesting someone to approve since Github is throwing internal error on me.

**Review by mohanarpit — APPROVED** [review]

**Review by trishaanand — APPROVED** [review]

**Pull request merged**